src/anomaly/automl_regressors.rs

```diff
@@ -0,0 +1,161 @@
+//! Meta-ML regressors built on the [automl] crate
+
+use automl::{
+    settings::{Algorithm, DecisionTreeRegressorParameters},
+    SupervisedModel,
+};
+use smartcore::dataset::Dataset;
+
+use super::metaml::{MetaMLDataset, MetaMLModel};
+
+// Transforms a MetaMLDataset into an automl dataset
+fn metaml_dataset_to_automl(data: MetaMLDataset) -> Dataset<f32, f32> {
+    // Double-check meta-information about the data
+    assert!(data.features.ncols() == 6);
```

Comment:
Seeing this again, I feel that the PR would benefit from a function that handled these assertions. Something like `assert_correct_shape`. That way, we could aggregate the TODO for compile-time shape checking into one place. Would make it easier to grep for the places we need to update.